### PR TITLE
Spyglass artifacts tab

### PR DESCRIPTION
**Pull request by tehcyx**
**Description**
Enables `gcsweb` deployment to be accessed via artifacts tab in Spyglass on Build results of a job.

Service is already deployed and working, this PR is the last change to enable the feature in the UI:
https://gcsweb.build.kyma-project.io/gcs/kyma-prow-logs/

Changes proposed in this pull request:
- config change to add link to build results
- deployments added to `cluster` folder

**Related issue(s)**
#1087

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -484,6 +484,7 @@ plank:
 deck:
   spyglass:
     size_limit: 500000000 # 500MB
+    gcs_browser_prefix: https://gcsweb.build.kyma-project.io/
```
Comment (michal-hudy) [inline prow/config.yaml]: It should probably points to `/gcs/`
```suggestion
    gcs_browser_prefix: https://gcsweb.build.kyma-project.io/gcs/
```

**Comment by bszwarc** [comment]
I would remove the entire `prow/cluster/gcsweb/README.md` document for the following reasons:
- the version mentioned in the Updating gcsweb section is also available [here](https://github.com/kyma-project/test-infra/blob/71f0ce8d543902cc42fb3d797b170746e03d92c8/prow/cluster/11-gcsweb.yaml)
- we may simply link to the original readme from here: https://github.com/kubernetes/k8s.io/tree/master/gcsweb.k8s.io 

**Suggestion**

Let's add a small section here: https://github.com/kyma-project/test-infra/blob/master/docs/prow/prow-architecture.md#components and call it `gcsweb` (similar to, for example, Branch Protector). 
The section will contain a simple overview and a link to further resources. For example:

### gcsweb

gcsweb is a lightweight web frontend for GCS which allows you to access the content of the **artifacts** tab in Spyglass without the need to log in. For more information on gcsweb read [this]( https://github.com/kubernetes/k8s.io/tree/master/gcsweb.k8s.io) document.

@michal-hudy , what do you think? do we need some more information to it?

File: prow/cluster/gcsweb/README.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
+# Info
```
Comment (bszwarc) [inline prow/cluster/gcsweb/README.md]: Let's remove this document and proceed this way:

Let's add a small section here: https://github.com/kyma-project/test-infra/blob/master/docs/prow/prow-architecture.md#components and call it `gcsweb` (similar to, for example, Branch Protector). 
The section will contain a simple overview and a link to further resources. For example:

### gcsweb

gcsweb is a lightweight web frontend for GCS which allows you to access the content of the **artifacts** tab in Spyglass without the need to log in. For more information on gcsweb read [this]( https://github.com/kubernetes/k8s.io/tree/master/gcsweb.k8s.io) document.

(as I posted in the comment)

**Comment by michal-hudy** [comment]
@bszwarc Yes, I agree small section will be much better than copied readme that we will need to maintain. Also, update architecture diagram 😄

File: prow/cluster/gcsweb/README.md (hunk 2 above)
Comment (tehcyx) [reply prow/cluster/gcsweb/README.md]: done

**Comment by tehcyx** [comment]
@michal-hudy @bszwarc I've removed the extra readme, added the paragraph and updated the architecture diagram.

**Comment by michal-hudy** [comment]
@tehcyx,  I fixed the flow in the diagram. Also, I created a follow-up issue for improving the readability of the diagram:: https://github.com/kyma-project/test-infra/issues/1149

**Pull request merged**